### PR TITLE
Added new test cases covering all edge scenarios for calc_std_and_verify

### PR DESCRIPTION
**Pull request by weknowthecalmat**
Added 15 test cases for - 

**Zero values** - test_zero_values
**Negative values** - test_negative_values_variance_mode
**NaN/Inf handling** - test_nan_values, test_inf_values
**Boundary testing** - test_exact_sigma_limit_matches, test_extreme_limits
**Error scenarios** - test_invalid_data_types, test_different_column_counts, test_empty_series
**Parameter matrix** - test_parameter_matrix_combinations
**Data validation** - test_empty_series, test_different_column_counts
**Numerical precision** - test_numerical_precision_small_values, test_numerical_precision_large_values, test_floating_point_edge_cases
**Additional robustness** - test_single_axis_dominance, test_variance_vs_std_dev_consistency

Relates to the issue - https://github.com/seafloor-geodesy/gnatss/issues/196

### CODE REVIEW

File: tests/ops/test_validate.py [hunk 1]
```diff
@@ -149,6 +150,301 @@ def test_calc_std_and_verify_with_variances(self):
...
+
+    def test_zero_values(self):
+        """Test handling of zero covariance values."""
+        zero_gps = pd.Series({
+            "ant_cov_XX1": 0.0,
+            "ant_cov_YY1": 0.0,
+            "ant_cov_ZZ1": 0.0,
+        })
+
+        result = calc_std_and_verify(zero_gps, std_dev=True, sigma_limit=0.05, verify=True)
```
Comment (johnbdesanto) [inline tests/ops/test_validate.py]: Only the second test is necessary, remove lines 163-164.

File: tests/ops/test_validate.py [hunk 2]
```diff
@@ -149,6 +150,301 @@ def test_calc_std_and_verify_with_variances(self):
...
+            "ant_cov_ZZ1": 0.01,
+        })
+
+        # Extremely small limit should fail
+        with pytest.raises(ValueError):
+            calc_std_and_verify(normal_gps, std_dev=True, sigma_limit=1e-10, verify=True)
+
+        # Extremely large limit should pass
+        result = calc_std_and_verify(normal_gps, std_dev=True, sigma_limit=1e10, verify=True)
+        assert result < 1e10
```
Comment (johnbdesanto) [inline tests/ops/test_validate.py]: Redundant, remove.

File: tests/ops/test_validate.py [hunk 3]
```diff
@@ -149,6 +150,301 @@ def test_calc_std_and_verify_with_variances(self):
...
+        # Test with all numeric data (should work)
+        numeric_gps = pd.Series({
+            "ant_cov_XX1": 0.01,
+            "ant_cov_YY1": 0.01,
+            "ant_cov_ZZ1": 0.01,
+        })
+
+        result = calc_std_and_verify(numeric_gps, std_dev=True, sigma_limit=0.05, verify=True)
+        expected = np.sqrt(0.01**2 + 0.01**2 + 0.01**2)
+        assert abs(result - expected) < 1e-10
```
Comment (johnbdesanto) [inline tests/ops/test_validate.py]: Redundant, remove.

File: tests/ops/test_validate.py [hunk 4]
```diff
@@ -149,6 +150,301 @@ def test_calc_std_and_verify_with_variances(self):
...
+        """Test cases where one axis dominates the 3D calculation."""
+        dominant_axis_gps = pd.Series({
+            "ant_cov_XX1": 0.05,  # Large value
+            "ant_cov_YY1": 1e-10,  # Tiny value
+            "ant_cov_ZZ1": 1e-10,  # Tiny value
+        })
+
+        result = calc_std_and_verify(dominant_axis_gps, std_dev=True, sigma_limit=0.1, verify=True)
+        # Result should be dominated by the X component
+        assert abs(result - 0.05) < 1e-6
```
Comment (johnbdesanto) [inline tests/ops/test_validate.py]: This test seems unnecessary, remove.

**Review by johnbdesanto — CHANGES_REQUESTED** [review]
I've left a number of comments. Many of the tests are similar to the point of testing the same things, so I have requested that those which are redundant be removed.

File: tests/ops/test_validate.py (hunk 3 above)
Comment (weknowthecalmat) [reply tests/ops/test_validate.py]: The intention was to test, invalid data fails, valid data works, within the same test. But as rightly said, this already is covered in the first test case, so I will remove it.

**Comment by weknowthecalmat** [comment]
@lsetiawan A part of it. So I wanted to verify that we cover all the possible combinations and we don't miss out on any. The 4 new test cases that I added as per suggestion from AI - test_parameter_matrix_combinations, test_numerical_precision_small_values / large_values, test_single_axis_dominance, test_extreme_limits. Others I validated my approach. I used GPT model  to brainstorm a bit for this. I think in this process this got a bit over-engineered, which I will avoid.

File: tests/ops/test_validate.py [hunk 5]
```diff
@@ -149,6 +150,301 @@ def test_calc_std_and_verify_with_variances(self):
...
+        result = calc_std_and_verify(large_gps, std_dev=True, sigma_limit=2e6, verify=True)
+        expected = np.sqrt(3 * (1e6)**2)
+        assert abs(result - expected) < 1e3  # Allow for some floating point error
+
+    def test_floating_point_edge_cases(self):
+        """Test floating-point arithmetic edge cases."""
+        # Test values that might cause floating-point precision issues
+        precision_gps = pd.Series({
+            "ant_cov_XX1": 0.1 + 0.2 - 0.3,  # Should be 0, but floating point...
+            "ant_cov_YY1": 1.0 / 3.0 * 3.0 - 1.0,  # Should be 0
```
[the remark this answers is not in the document]
Comment (weknowthecalmat) [reply tests/ops/test_validate.py]: Hi @johnbdesanto , I was wondering if this is actually required as we are testing it implicitly.

**Comment by weknowthecalmat** [comment]
@johnbdesanto I have added the changes as per the review feedback
